### PR TITLE
Intl-ICU: Merge branch `intl-icu`: Intl.NumberFormat and node-cc build integration into branch `release/1.7`

### PR DESCRIPTION
**Pull request by dilijev**
Includes the following PRs merged to branch `intl-icu`

* [MERGE #3809 @dilijev] Intl-ICU: implement Intl.NumberFormat under ICU
* [MERGE #3820 @jackhorton] Implements ICU version of getDefaultLocale, fixes up ICU extension handling
* [MERGE #3822 @obastemur] xplat: fix the path for i18n
* [MERGE #3750 @jackhorton] Expose platform.winglob to Intl.js to detect which i18n lib we are using in the native code.

### CODE REVIEW

File: lib/Runtime/Library/InJavascript/Intl.js [hunk 1]
```diff
@@ -862,6 +975,22 @@
 
     // Intl.NumberFormat, Number.prototype.toLocaleString
     var NumberFormat = (function () {
+        // Keep these "enums" in sync with lib/Runtime/PlatformAgnostic/Intl.h
+        const NumberFormatStyle = setPrototype({
```
Comment (boingoing) [inline lib/Runtime/Library/InJavascript/Intl.js]: Nice enums 👍

File: lib/Runtime/Library/InJavascript/Intl.js [hunk 2]
```diff
@@ -7,13 +7,23 @@
 // Core intl lib
 (function (EngineInterface, InitType) {
     var platform = EngineInterface.Intl;
+
+    // allow unit tests to disable caching behavior for testing convenience but have this always `true` in real scenarios
+    platform.useCaches = true;
```
Comment (boingoing) [inline lib/Runtime/Library/InJavascript/Intl.js]: I don't see how test code can set this?

Comment (jackhorton) [reply lib/Runtime/Library/InJavascript/Intl.js]: It is used in the "IntlPlatformTests" chakrafull unit test

File: lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+        bool success = false;
+        UErrorCode error = UErrorCode::U_ZERO_ERROR;
+        char icuLocaleId[ULOC_FULLNAME_CAPACITY] = { 0 };
+        char icuLangTag[ULOC_FULLNAME_CAPACITY] = { 0 };
+        int32_t parsedLength = 0;
+        int32_t forLangTagResultLength = 0;
+        int32_t toLangTagResultLength = 0;
+        size_t inputLangTagUtf8SizeActual = 0;
+
+        // Allocate memory for the UTF8 output buffer. Need 3 bytes for each (code point + null) to satisfy SAL.
```
Comment (MSLaguana) [inline lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp]: Is there a reason you avoided the `WideToNarrow` class in `Utf8Helper.h` and did your own memory allocation / utf conversion calls?

File: lib/Runtime/Library/InJavascript/Intl.js (hunk 2 above)
Comment (dilijev) [reply lib/Runtime/Library/InJavascript/Intl.js]: Through the Debug object (exposed through a test flag) which allows tests to manipulate the EngineInterfaceObject passed to Intl code. At some point we should migrate that test option and associated tests to ChakraCore.

File: lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp (hunk 3 above)
Comment (dilijev) [reply lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp]: No particular reason. Wasn't aware, I guess. I'll look into it and we can update the implementation later.

**Pull request merged**